framework/doc/content/documentation/systems/AuxKernels/VectorPostprocessorVisualizationAux.md

```diff
@@ -0,0 +1,23 @@
+# VectorPostprocessorVisualizationAux
+
+## Short Description
+
+!syntax description /AuxKernels/VectorPostprocessorVisualizationAux
+
+## Description
+
+This object is intended to let you view VectorPostprocessor vectors that are of lenght `num_procs` (meaning there is one value per MPI process).  This object will take those values and fill up an Auxiliary field with them so the values can be visualized.
```

Comment:
I guess as a utility this is fine, but there are better ways to accomplish this. If you take a step back and look at these new utilities you've added in the past week, you are essentially taking local only information, replicating it on all processors, then using the local only information. We could have just never sent it in the first place if you are _only_ using the AuxKernel, but I suppose you may want both the VPP information and the visualization field.

Reply:
Ok - can you enumerate the "better ways"?  And I'd like to see some citations on `these new utilities you've added in the past week, you are essentially taking local only information, replicating it on all processors, then using the local only information`

Where?

First of all - I didn't `replicate information to all` at all with any of my previous utilities.  WorkBalance, HistogramVectorPostprocessor and StatisticsVectorPostprocessor only compute final values on processor zero (by default).

Besides that - HistorgramVectorPostprocessor and StatisticsVectorPostprocessor works on _any_ VPP... not just VPPs that compute one value per processor.  For instance: you can use it to compute the mean, max, sum, average or get the Histogram of values found by a LineValueSampler... including the statistics/historgram for _all_ of the variables you sampled by that LineValueSampler.  Please let me know how else you would do this.

I will agree that in this case - for WorkBalance - it needs to broadcast to get a copy of the information on each processor - but there might be other reasons for doing that besides just visualizing it.

What I'm making here are _basic_ tools.  I have many other plans for them going forward with my dissertation.  If you don't find them useful then I'll just put them in my app.  I _know_ that many of these will get used a TON by many people over the years.

Reply:
Also: please tell me how to get the partition surface area on each processor and paint it into a field _only_ using an AuxKernel?

Reply:
👍 resolved